src/Service/Authorization/AuthorizationResolver.cs

```diff
@@ -648,10 +659,22 @@ public IEnumerable<string> GetRolesForOperation(string entityName, Config.Operat
         /// <param name="entityName">EntityName whose operationMetadata will be searched.</param>
         /// <param name="field">Field to lookup operation permissions</param>
         /// <param name="operation">Specific operation to get collection of roles</param>
-        /// <returns>Collection of role names allowed to perform operation on Entity's field.</returns>
+        /// <returns>Collection of role names allowed to perform operation on Entity's field. Empty list when zero roles
+        /// have permission to perform the {operation} on the provided field.</returns>
         public IEnumerable<string> GetRolesForField(string entityName, string field, Config.Operation operation)
         {
-            return EntityPermissionsMap[entityName].FieldToRolesMap[field][operation];
```

Comment:
The assumption was this would be validated in the runtime config validator or every field will be added to this `FieldToRolesMap`. Is this assumption invalid due to the presence of relationship fields?

Reply:
I identified this as an uncaptured edge case when there does not exist a role defined on an entity that "allows" the field. FieldToRolesMap is populated with "allowed" fields. This situation occurs during schema creation when attempting to create an authorize directive for the field. 
For example, if an entity has one role and includes all but one field: 'excludedField', that excluded field won't be present in the FieldToRolesMap and will fail schema creation when that field is not found (Key not found error). 
This case is unique to GraphQL schema creation and not runtime execution because once a schema is successfully built, all input fields and object type definitions (in this use case), do not reference that excluded field as expected, thus avoiding any lookup of that field in the FieldToRolesMa, because any user usage of that field would result in a query validation issue prior to resolver execution.

Reply:
I believe i did this out of abundance of caution. I couldnt identify a case this fails after i finalized for publishing the pr.

Reply:
Good catch of edge case during schema creation!